src/components/AttendantForm.jsx

```diff
@@ -0,0 +1,191 @@
+              e.target.setCustomValidity("Please enter a valid name")
+            }
+            onInput={(e) => e.target.setCustomValidity("")}
+          />
+        </div>
+        <br />
+        <div className="input">
+          <label htmlFor="lastName">Enter last name:</label>
+          <input
+            type="text"
+            minLength="2"
+            maxLength="50"
+            pattern="[A-Za-zÀ-ÖØ-öø-ÿ'\-]+"
+            name="lastName"
+            placeholder="Last Name"
+            data-testid="last-name"
+            value={lastName}
+            required
+            onChange={(e) => {
+              setLastName(e.target.value);
+            }}
+            onInvalid={(e) =>
+              e.target.setCustomValidity("Please enter a valid last name")
+            }
+            onInput={(e) => e.target.setCustomValidity("")}
+          />
+        </div>
+        <br />
+        <div className="input">
+          <label htmlFor="jobTitle">Select job title:</label>
+          <select
+            name="jobTitle"
+            data-testid="job-title"
+            required
+            value={jobTitle}
+            onChange={(e) => setJobTitle(e.target.value)}
+            onInput={(e) => e.target.setCustomValidity("")}
+            onInvalid={(e) =>
+              e.target.setCustomValidity("Please select a job title")
+            }
+          >
+            <option
+              value=""
+              style={{ textAlign: "center" }}
+            >
+              {isLoading
+                ? "Loading job titles... 🔄"
+                : "Please select a job title ⬇️"}
+            </option>
+            {allJobTitles.map((jobTitle) => (
+              <option
+                key={jobTitle}
+                value={jobTitle}
+              >
+                {jobTitle}
+              </option>
+            ))}
+          </select>
+        </div>
+        <br />
+        <div className="input">
+          <label htmlFor="age">Enter age:</label>
+          <input
+            type="number"
+            min="15"
+            max="120"
+            name="age"
+            placeholder="Age"
+            data-testid="age"
+            value={age}
+            required
+            onChange={(e) => {
+              setAge(e.target.value);
+            }}
+            onInvalid={(e) =>
+              e.target.setCustomValidity("Age range must be 15-120")
+            }
+            onInput={(e) => e.target.setCustomValidity("")}
+          />
+        </div>
+        <br />
+        <div className="button">
+          <button
+            type="submit"
+            data-testid="submit"
+            disabled={isLoading}
+          >
+            Submit
+          </button>
+          {isLoading && <div>Loading...</div>}
+        </div>
+      </form>
+      {attendants.map((e) => (
+        <AttendantDisplay
+          key={`${e.name}${e.lastName}${e.jobTitle}${e.age}`}
+          attendant={e}
+        />
+      ))}
```

Comment:
Since attendants do not have ids, an index can be used and shortened. Also the mapping function can be extracted to a `renderAttendants` function before the `return`.
```suggestion
      {attendants.map((attendant, index) => (
        <AttendantDisplay
          key={`attendant-${index}`}
          attendant={attendant}
        />
      ))}
```